testing/file_test/autoupdate.cpp

```diff
@@ -222,30 +236,30 @@ auto AutoupdateFileTest(
 
   // Stitch together content.
   llvm::SmallVector<const FileTestLineBase*> new_lines;
-  for (auto [filename, non_check_file, stdout_check_file, stderr_check_file] :
-       llvm::zip(filenames, non_check_lines, stdout_check_lines,
-                 stderr_check_lines)) {
+  for (auto [file_number, filename, non_check_file] :
+       llvm::enumerate(filenames, non_check_lines)) {
     llvm::DenseMap<int, int> output_line_remap;
+    llvm::SmallVector<CheckLine*> check_lines_this_file;
     int output_line_number = 0;
-    auto* stdout_check_line = stdout_check_file.begin();
-    auto* stderr_check_line = stderr_check_file.begin();
 
     // Add all check lines from the given vector until we reach a check line
     // attached to a line later than `to_line_number`.
-    auto add_check_lines = [&](const llvm::SmallVector<CheckLine>& lines,
+    // Explicit capture of file_number is a workaround for the inability for
+    // lambdas to implicitly capture a structured binding.
+    auto add_check_lines = [&, file_number = static_cast<int>(file_number)](
```

Comment:
```suggestion
    // Work around the inability to implicitly capture a structured binding.
    auto file_number = file_number;
    auto add_check_lines = [&](
```

Just mulling, does this work (and do you find it any easier to read)?

Reply:
No, that doesn't work, the variable is in scope in its own initializer. We would need to give it a different name, at least.

I've added a couple of suggestions, take  your pick :-)